### DISCUSSION TITLE
Iteration documentation and it-expansion

### DISCUSSION THREAD

**Discussion by aborruso** [nushell/nushell#8414] [Q&A] [answered]
Hi,
I'm trying to build a simple task: remove all items created more than 60 days ago.

And I have written this

```
ls **\*  -l |  where created <= (date now) - 60day | each { |it| rm $it.name }
```

It seems to work.

I'm using `0.74.0`  and I didn't understand if it's still possible to use [this old simpler syntax](https://www.nushell.sh/blog/2020-05-12-nushell_0_14_0.html#it-expansion-jonathandturner-thegedge) with auto-expand, in which `$it` expands itself to a full `each` call.

What is the documentation page where i can read about this?

I'm confused because I can run `ls | get name`, but not `ls | rm name` or `ls | each { rm name}`, or `ls | each { rm $it.name}`.

I have tried also 

```
ls | where size == 0KB and type == file | each { rm $in.name } | null
```

but it does not delete my 0KB files.

Thank you and sorry for these stupid questions.

**Comment by aborruso** [comment]
> I have tried also
> 
> ```
> ls | where size == 0KB and type == file | each { rm $in.name } | null
> ```

Ok If I remove `| null` it works. But now I have some questions.

Why in the documentation I have this example and it does not work on my PC?

![image](https://user-images.githubusercontent.com/30607/224535980-66958e3f-dfee-443b-874c-8d24fd7efd9b.png)

Since I can't use `null`, how to not have errors if the list is empty?

I have found notes about `$in` in [Basic page](https://www.nushell.sh/book/pipelines.html#basics). 
Is there any other `$in` dedicated page? In any case, this page answers some of my doubts.

Thank you

**Reply by fdncred** [reply]
We've changed things a bit recently. I think the `| null` part should be removed in that command since it works fine without it.

**Reply by aborruso** [reply]
> We've changed things a bit recently. I think the `| null` part should be removed in that command since it works fine without it.

Thank you. But is there a way to do not have an error, if I have no output from ls?

If this is a daily cron job on a folder, tomorrow I could have no result from `ls`. And then if I will run

```
ls | where size == 0KB and type == file | each { rm $in.name }
```

I will have something like this

```
  × Input type not supported.
   ╭─[entry #6:1:1]
 1 │ ls | where size == 0KB and type == file | each { rm -f $in.name }
   · ─┬   ──┬──
   ·  │     ╰── only list, binary, raw data or range input data is supported
   ·  ╰── input type: nothing
   ╰────
```

Thank you

**Reply by fdncred** [reply]
In the latest main, if I run this `ls | where size == 0KB and type == file | each { rm $in.name }` 5 times in a row, there are no errors. Is that what you mean?

**Reply by aborruso** [reply]
> In the latest main, if I run this `ls | where size == 0KB and type == file | each { rm $in.name }` 5 times in a row, there are no errors. Is that what you mean?

I'm using `0.74.1`. If I run `ls | where size == 0KB and type == file` in an empty folder I have this error.
Do I have to update my nushell?

Thank you

![image](https://user-images.githubusercontent.com/30607/224741596-4049aec7-816a-4470-9e30-b03aad0aae48.png)

**Reply by fdncred** [reply]
yes. I'm testing on 0.76.1. 0.77.0 releases tomorrow sometime so you might want to wait for 77.

**Reply by aborruso** [reply]
> yes. I'm testing on 0.76.1. 0.77.0 releases tomorrow sometime so you might want to wait for 77.

I have the same error using `0.76.0`. I will wait for the next release.
Thank you

**Comment by aborruso** [comment, marked as answer]
@fdncred I have had a satisfactory answer in discord: to use try and catch

```
try {ls | where size == 0KB and type == file | each { rm $in.name }} catch { echo 'no ls output' }
```
